src/script/lua_api/l_object.cpp

```diff
@@ -729,16 +734,25 @@ int ObjectRef::l_set_properties(lua_State *L)
 	ObjectRef *ref = checkobject(L, 1);
 	ServerActiveObject *co = getobject(ref);
 	if (co == NULL) return 0;
+
 	ObjectProperties *prop = co->accessObjectProperties();
 	if (!prop)
 		return 0;
 	read_object_properties(L, 2, prop, getServer(L)->idef());
+
 	if (prop->hp_max < co->getHP()) {
 		PlayerHPChangeReason reason(PlayerHPChangeReason::SET_HP);
 		co->setHP(prop->hp_max, reason);
 		if (co->getType() == ACTIVEOBJECT_TYPE_PLAYER)
 			getServer(L)->SendPlayerHPOrDie((PlayerSAO *)co, reason);
 	}
+
+	if (co->getType() == ACTIVEOBJECT_TYPE_PLAYER) {
+		PlayerSAO *playersao = getplayersao(ref);
+		if (prop->breath_max < playersao->getBreath())
+			playersao->setBreath(prop->breath_max);
```

Comment:
Move this code to `PlayerSAO::setBreath`; range limit `breath` there.
https://github.com/minetest/minetest/blob/ae1caba6aaa6f0a13e70f195f371fd4e7a378491/src/content_sao.cpp#L1343

Reply:
Done